plugin/storage/es/spanstore/writer.go

```diff
@@ -149,6 +151,14 @@ func (s *SpanWriter) Close() error {
 
 func (s *SpanWriter) createIndex(indexName string, mapping string, jsonSpan *dbmodel.Span) error {
 	if !keyInCache(indexName, s.indexCache) {
+		s.indexMutex.Lock()
+		defer s.indexMutex.Unlock()
+
+		// re-check if index exists in case other goroutine did the job under lock for us
+		if keyInCache(indexName, s.indexCache) {
+			return nil
+		}
```

Comment:
any way to add a test for this, e.g. to ensure that s.client.IndexExists() is not called twice (via mock)